0xProject/0x-api
# feat: Create rfqm worker heartbeat report

# Description


Creates entity definition, migrations, and implements the worker heartbeat report. As implemented, the worker makes a report right before it starts its long polling for new messages.

# Testing Instructions


I went through some trouble to make sure the database updates the timestamp with each new report. ~~There's a unit test implemented to check this.~~ I went through a lot of work to try to test this with a db-level unit test, but our test db is set up through TypeORM's `synchronize` functionality, whereas the timestamp trigger depends on running the full migration. After some tinkering, I don't think the extra effort is worth it and I've elected to test this manually.

To manually test: Setup test DB and run migrations. Spin up worker runner with a worker that has a mnemonic, worker index, and some balance in the associated address. Ensure the worker creates a report before each message polling cycle, and ensure subsequent reports have an updated timestamp.

# Checklist





    For more information see `0x API Matcha smoke test runbook` in Quip.
